src/crush/CrushWrapper.cc

```diff
@@ -2347,7 +2347,12 @@ int CrushWrapper::_choose_type_stack(
       underfull_buckets[j].insert(item);
     }
   }
-  ldout(cct, 20) << __func__ << " underfull_buckets " << underfull_buckets << dendl;
+  ldout(cct, 20) << __func__ << " underfull_buckets " << "[";
+  for (auto p = underfull_buckets.begin(); p != underfull_buckets.end(); ++p) {
```

Comment:
>   if (&obj != underfull_buckets.begin())
aah! sorry my bad. As kefu specified above, it can be compared with the object. But ignore this change as  it will make it complicated, not simpler.